plugin/evm/testutils/testutils_test.go

```diff
@@ -0,0 +1,138 @@
+package testutils
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestMustNotImport fails if:
+// - a package in plugin/evm imports the testutils package
+// - a non-test file imports the testutils package
+func TestMustNotImport(t *testing.T) {
+	prodOnly := false
+	checkNotImporting(t, "../", prodOnly, "testutils should only be used outside plugin/evm")
```

Comment:
why do we have such limitation? Does that mean we can't use this pkg in plugin/evm test files?

Reply:
See [this comment](https://github.com/ava-labs/subnet-evm/pull/1422#discussion_r1916368549). TLDR:

- this testutils package is only to limit diffs in geth code
- (subjective) we should prefer to copy paste test helping utilities, to reduce package dependencies and possible cycle import issues

Reply:
I don't think we should limit this pkg like that. if we nee such a pkg it should be in internal if possible and under a different pkg than plugin/evm. I think it should be sufficient to check this is not imported in any prod code, no?

I'm fine with copy pasting `WithMetrics` if we feel like it's best not to import `metrics` here.

Reply:
thinking again, having a `testutils` package actually indicates this should be not called from prod code. So not sure if there is any point to strict that down too much.

Reply:
Moved to internal/testutils, without import restrictions